x-pack/legacy/plugins/siem/public/components/matrix_histogram/index.tsx

```diff
@@ -4,92 +4,185 @@
-import { EuiLoadingContent } from '@elastic/eui';
+import { EuiLoadingContent, EuiSelect } from '@elastic/eui';
+import { noop } from 'lodash/fp';
+import * as i18n from './translations';
 import { BarChart } from '../charts/barchart';
 import { HeaderSection } from '../header_section';
-import { ChartSeriesData } from '../charts/common';
 import { DEFAULT_DARK_MODE } from '../../../common/constants';
 import { useUiSetting$ } from '../../lib/kibana';
 import { Loader } from '../loader';
 import { Panel } from '../panel';
-import { getBarchartConfigs, getCustomChartData } from './utils';
-import { MatrixHistogramProps, MatrixHistogramDataTypes } from './types';
+import { getBarchartConfigs, getCustomChartData } from '../../components/matrix_histogram/utils';
+import { useQuery } from '../../containers/matrix_histogram/utils';
+import {
+  MatrixHistogramProps,
+  MatrixHistogramOption,
+  HistogramAggregation,
+  MatrixHistogramQueryProps,
+} from './types';
+import { generateTablePaginationOptions } from '../paginated_table/helpers';
+import { ChartSeriesData } from '../charts/common';
 
-export const MatrixHistogram = ({
-  data,
-  dataKey,
-  endDate,
-  id,
-  loading,
-  mapping,
-  scaleType = ScaleType.Time,
-  startDate,
-  subtitle,
-  title,
-  totalCount,
-  updateDateRange,
-  yTickFormatter,
-  showLegend,
-}: MatrixHistogramProps<MatrixHistogramDataTypes>) => {
-  const barchartConfigs = getBarchartConfigs({
-    from: startDate,
-    to: endDate,
-    onBrushEnd: updateDateRange,
-    scaleType,
-    yTickFormatter,
+export const MatrixHistogram = React.memo(
+  ({
+    activePage,
+    alertsType,
+    anomaliesType,
+    authenticationsType,
+    dataKey,
+    defaultStackByOption,
+    endDate,
+    eventsType,
+    filterQuery,
+    hideHistogramIfEmpty = false,
+    id,
+    isPtrIncluded,
+    isInspected,
+    legendPosition,
+    limit,
+    mapping,
+    query,
+    scaleType = ScaleType.Time,
+    setQuery,
     showLegend,
-  });
-  const [showInspect, setShowInspect] = useState(false);
-  const [darkMode] = useUiSetting$<boolean>(DEFAULT_DARK_MODE);
-  const [loadingInitial, setLoadingInitial] = useState(false);
+    skip,
+    stackByOptions,
+    startDate,
+    subtitle,
+    title,
+    updateDateRange,
+    yTickFormatter,
+    sort,
+  }: MatrixHistogramProps & MatrixHistogramQueryProps) => {
+    const barchartConfigs = getBarchartConfigs({
+      from: startDate,
+      to: endDate,
+      onBrushEnd: updateDateRange,
+      scaleType,
+      yTickFormatter,
+      showLegend,
+      legendPosition,
+    });
+    const [showInspect, setShowInspect] = useState(false);
+    const [darkMode] = useUiSetting$<boolean>(DEFAULT_DARK_MODE);
 
-  const barChartData: ChartSeriesData[] = getCustomChartData(data, mapping);
+    const handleOnMouseEnter = useCallback(() => {
+      if (!showInspect) {
+        setShowInspect(true);
+      }
+    }, [showInspect]);
```

Comment:
consider declaring `setShowInspect ` as a dependency here

Reply:
fwiw, i don't expect the reference to `setShowInspect`, but given that's an implementation detail of hooks, i prefer to include all of the deps used within `useCallback`